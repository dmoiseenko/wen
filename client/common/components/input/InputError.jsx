import React from 'react';
import PropTypes from 'prop-types';


export default function InputErrors({ error }) {
  if (!error) {
    return null;
  }

  return (
    <p className="help is-danger">{error}</p>
  );
}

InputErrors.propTypes = {
  error: PropTypes.string.isRequired
};

InputErrors.defaultProps = {
  error: ''
};
