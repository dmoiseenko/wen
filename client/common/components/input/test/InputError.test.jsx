import React from 'react';
import renderer from 'react-test-renderer';

import InputError from '../InputError';


it('should render', () => {
  const props = {
    error: 'error'
  };

  const tree = renderer.create(
    <InputError {...props} />
  );

  expect(tree.toJSON()).toMatchSnapshot();
});

it('should not render if no error', () => {
  const props = {};

  const tree = renderer.create(
    <InputError {...props} />
  );

  expect(tree.toJSON()).toMatchSnapshot();
});
