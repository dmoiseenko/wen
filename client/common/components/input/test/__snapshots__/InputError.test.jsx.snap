// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render if no error 1`] = `null`;

exports[`should render 1`] = `
<p
  className="help is-danger"
>
  error
</p>
`;
